paper/paper.md

```diff
@@ -0,0 +1,114 @@
+---
+title: '`optimade-python-tools`: a Python library for implementing and consuming materials data via OPTIMADE APIs'
+tags:
+  - Python
+  - REST API
+  - JSON:API
+  - crystallography
+  - density-functional theory
+  - ab initio
+  - materials discovery
+  - databases
+authors:
+  - name: Matthew L. Evans
+    orcid:  0000-0002-1182-9098
+    affiliation: "1, 2"
+  - name: Casper Andersen
+    orcid: 0000-0002-2547-155X
+    affiliation: 3
+  - name: Shyam Dwaraknath
+    orcid: 0000-0003-0289-2607
+    affiliation: 4
+  - name: Markus Scheidgen
+    orcid: 0000-0002-8038-2277
+    affiliation: 5
+  - name: Ádám Fekete
+    orcid: 0000-0002-6263-897X
+    affiliation: "1, 6, 8"
+  - name: Donald Winston
+    orcid: 0000-0002-8424-0604
+    affiliation: "4, 7"
+affiliations:
+ - name: Institut de la Matière Condensée et des Nanosciences, Université catholique de Louvain, Chemin des Étoiles 8, Louvain-la-Neuve 1348, Belgium
+   index: 1
+ - name: Theory of Condensed Matter Group, Cavendish Laboratory, University of Cambridge, J. J. Thomson Avenue, Cambridge, CB3 0HE, U.K.
+   index: 2
+ - name: EPFL
+   index: 3
+ - name: Lawrence Berkeley National Laboratory, Berkeley, CA, USA
+   index: 4
+ - name: FHI
+   index: 5
+ - name: Namur
+   index: 6
+ - name: Polyneme LLC
+   index: 7
+ - name: KCL
+   index: 8
+
+date: June 2021
+bibliography: paper.bib
+---
+
+# Summary
+
+<!--Follow similar spiel to OPTIMADE paper:-->
+<!--- advent of high-throughput computing, software and theory maturity, availability of compute power have lead to explosion of computational data.-->
+<!--- can be directly compared to high-quality measurements of crystal structures curated over many years-->
+<!--- this data is increasingly being made available via public APIs, such as...-->
+<!--- The OPTIMADE API specification was created to enable interoperability and machine-actionable APIs from multiple data providers-->
+
+In recent decades, improvements in algorithms, hardware, and theory have enabled crystalline materials to be studied at the atomistic level with great accuracy and speed.
+To enable dissemination, reproducibility, and reuse, many digital crystal structure databases have been created and curated, ready for comparison with existing infrastructure storing structural characterizations of real crystals.
```

Comment:
```suggestion
To enable dissemination, reproducibility, and reuse, many digital crystal structure databases have been created and curated, ready for comparison with existing infrastructure that stores structural characterizations (e.g., diffraction) of real crystals.
```

Reply:
Thanks @ml-evs for the work you put already in there. 

I think there could be more emphases on "use-cases". You rightly dissect the functionality into routing, filter-language, db/resource-handling. I think we could find a few examples of how to recombine those functions in multiple abstract scenarios rather than "simply" stating what MP, MC, NOMAD are doing? I feel readers would identify more with more general use-cases than with our specifics?
- use OPT as a standalone app, e.g. to server your data
- integrate it into an existing app, e.g. on top of an existing database backend [e.g. what MP, MC, NOMAD are doing]
- use the filter independent of an app and routing (e.g. NOMAD uses it in our UI/API searches)
- maybe even more novel ideas like this "run a auto-configured OPT docker container" directly on your data

Reply:
Could you remind me on the scope of the paper? I assume it addresses ML, data science people? Is this a physics or materials science specific community? What are you page boundaries?

Reply:
Hi @markus1978, sorry for missing this earlier.

> I think there could be more emphases on "use-cases". You rightly dissect the functionality into routing, filter-language, db/resource-handling. I think we could find a few examples of how to recombine those functions in multiple abstract scenarios rather than "simply" stating what MP, MC, NOMAD are doing? I feel readers would identify more with more general use-cases than with our specifics?

I think you are probably right, and I like your suggestions. Of course, these are things we said we would write anyway :)

Reply:
> Could you remind me on the scope of the paper? I assume it addresses ML, data science people? Is this a physics or materials science specific community? What are you page boundaries?

JOSS is multi-disciplinary, but purely focused on the open software. The review will be done a software developer with expertise in matsci (basically like us), and I find that they focus more on the repository than the contents of the paper itself (compared to a scientific journal). There are no page boundaries but the papers are typically short (1-4 pages) that acts as adverts for the code itself.